core/src/main/scala/org/bitcoins/core/protocol/ln/LnInvoiceTags.scala

```diff
@@ -105,23 +174,23 @@ sealed abstract class LnInvoiceTags {
 
   def fallbackAddress: Option[LnInvoiceTag.FallbackAddressTag]
 
-  def routingInfo: Option[Vector[LnRoutingInfo]] //TODO: Not implemented
+  def routingInfo: Option[LnInvoiceTag.RoutingInfo]
 
   override def toString: String = {
     val empty = ""
 
-    val b = new StringBuilder()
+    val b = new mutable.StringBuilder()
 
     b.append(paymentHash.toString)
-    b.append(description.getOrElse(empty))
-    b.append(spk.getOrElse(empty))
-    b.append(descriptionHash.getOrElse(empty))
-    b.append(expiryTime.getOrElse(empty))
-    b.append(expiryTime.getOrElse(empty))
-    b.append(cltvExpiry.getOrElse(empty))
-    b.append(fallbackAddress.getOrElse(empty))
+    b.append(description.map(_.toString).getOrElse(empty))
+    b.append(nodeId.map(_.toString).getOrElse(empty))
+    b.append(descriptionHash.map(_.toString).getOrElse(empty))
+    b.append(expiryTime.map(_.toString).getOrElse(empty))
+    b.append(cltvExpiry.map(_.toString).getOrElse(empty))
+    b.append(fallbackAddress.map(_.toString).getOrElse(empty))
 
-    //TODO: add routing info
+    val routes = routingInfo.map(_.toString).getOrElse(empty)
```

Comment:
Is there a reason these two lines aren't just combined into `b.append(routingInfo.map(_.toString).getOrElse(empty))`?

Reply:
It gets a little long. I like using intermediate `vals` if the thing we are passing to a method is not trivial.